surrogates/common_utils/fits.py

```diff
@@ -0,0 +1,139 @@
+##==============================================================================
+## BHPTNRSurrogate module
+## Description : evaluates the (spline or GPR) surrogate model
+## Author : Tousif Islam, Nov 2022 [tislam@umassd.edu / tousifislam24@gmail.com]
+##==============================================================================
+
+import numpy as np
+import scipy
+from scipy.interpolate import splrep, splev
+from . import utils
+from .eval_pysur import evaluate_fit as evaluate_GPR
+
+#----------------------------------------------------------------------------------------------------
+def evaluate_GPR_at_EIM_nodes(X, fit_data):
+    """ Evaluate the spline at one EIM node """
+    return None
+
+#----------------------------------------------------------------------------------------------------
+def evaluate_splines_at_EIM_nodes(X, fit_data):
+    """ Evaluate the spline at one EIM node """
+    
+    [h_eim_spline, eim_indicies] = fit_data
+    return np.array([splev(X, h_eim_spline[j]) for j in range(len(eim_indicies))])
+
+
+#----------------------------------------------------------------------------------------------------
+def EIM_B_to__waveform_datapiece(B, eim_vals):
+    """ Compute the interpolated waveform for a single mode : dominant 22 mode """
+    
+    approx_datapiece = np.dot(B.transpose(), eim_vals)
+    return approx_datapiece
+
+
+#----------------------------------------------------------------------------------------------------
+def evaluate_datapiece(X, fit_data, B, fit_func):
+    """ Compute the datapiece for the input parameters"""
+    
+    # evaluates spline fits at eim nodes
+    if fit_func == 'spline_1d':
+        h_eim_datapiece = evaluate_splines_at_EIM_nodes(X, fit_data)
+    # evaluates GPR fits at eim nodes
+    elif fit_func == 'GPR_fits':
+        h_eim_datapiece = evaluate_GPR_at_EIM_nodes(X, fit_data)
+    # combine h_eim and  eim basis matrix to give full datapiece
+    h_approx_datapiece = EIM_B_to__waveform_datapiece(B, h_eim_datapiece) 
+    
+    return h_approx_datapiece
+
+
+#----------------------------------------------------------------------------------------------------
+def evaluate_surrogate_mode(X, fit_data_1, fit_data_2, B_datapiece_1, B_datapiece_2, 
+                            fit_func, decomposition_func, norm):
+    """ Compute the interpolated waveform for a single mode : higher order modes"""
```

Comment:
I think more documentation is needed to explain what this function is doing. In particular, I am confused about...

1. does this function not work for the 22 mode? If not, why not? What routine is used to compute the 22 mode?
2. this looks like an "internal" function that isn't supposed to be called outside of all_modes_surrogate. As such, I would suggest adding an underscore to the function, so _evaluate_surrogate_mode. This will make it clear that its used for internal stuff (ie its a "private" function). This is used especially for private methods of classes, but I think its helpful here too (https://peps.python.org/pep-0008/#method-names-and-instance-variables). Other functions in this python file that shouldn't be called directly by the user could also get an underscore.
3. I would mention in the documentation that the user should look at all_modes_surrogate for information on the inputs.

Reply:
Great! Changes made.